lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1470,12 +1471,20 @@ static LogicalResult canonicalizeSingleSetConnect(StrictConnectOp op,
     }
   }
 
-  // Replace all things *using* the decl with the constant/port, and
-  // remove the declaration.
-  rewriter.replaceOp(connectedDecl, replacement);
+  if (isUselessName(connectedDecl)) {
+    // Replace all things *using* the decl with the constant/port, and
+    // remove the declaration.
+    rewriter.replaceOp(connectedDecl, replacement);
 
-  // Remove the connect.
-  rewriter.eraseOp(op);
+    // Remove the connect
+    rewriter.eraseOp(op);
+  } else {
+    // bypass the decl, but keep it around
+    rewriter.startRootUpdate(connectedDecl);
+    connectedDecl->replaceAllUsesWith(ArrayRef{replacement});
+    op->setOperand(0, connectedDecl->getResult(0));
```

Comment:
The mutation on `connectedDecl` is being registered with the rewriter here, but this also mutates `op`. Does this need something like `rewriter.updateRootInPlace(op, [](){ op->setOperand(...) }` to be squeeky clean?